lib/src/home/contacts/add_contacts_page.dart

```diff
@@ -0,0 +1,237 @@
+      },
+      decoration: const InputDecoration(
+        labelText: Strings.phoneNumberLabel,
+        icon: Icon(Icons.phone),
+      ),
+    );
+
+    final picture = SizedBox(
+        width: 120.0,
+        height: 120.0,
+        child: GestureDetector(
+          onTap: () async {
+            showDialog(
+                context: context,
+                builder: (BuildContext context) =>
+                    ImageChooserDialog(onImagePick: (image) {
+                      setState(() {
+                        _contactPhoto = image;
+                      });
+                    }));
+          },
+          child: CircleAvatar(
+              child: ImageWithPlaceholderIcon(
+                  key: Keys.avatarKey, image: _contactPhoto)),
+        ));
+
+    ListView content = ListView(
+      padding: const EdgeInsets.symmetric(vertical: 20.0, horizontal: 10.0),
+      children: <Widget>[
+        const SizedBox(height: 20),
+        Form(
+          key: _formKey,
+          child: Column(
+            children: <Widget>[
+              picture,
+              inputFirstName,
+              inputLastName,
+              inputPhoneNumber
+            ],
+          ),
+        ),
+      ],
+    );
+
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.close),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        ),
+        title: const Text(Strings.addContactPageTitle),
+        actions: <Widget>[
+          SizedBox(
+            width: 80,
+            child: IconButton(
+              icon: const Text(
+                'Save',
+                style: TextStyle(fontWeight: FontWeight.bold),
+              ),
+              onPressed: _submitFormAndPopBack,
+            ),
+          )
+        ],
+      ),
+      body: content,
+    );
+  }
+
+  /// creates a contact by validating and submitting form data;
+  /// pop back to previous screen
+  void _submitFormAndPopBack() async {
+    if (_formKey.currentState?.validate() == true) {
+      PermissionStatus permissionStatus = await _getContactPermission();
+      if (permissionStatus.isGranted) {
+        ServiceLocator.instance.contactsRepository.addContact(
+            firstName: _contactFirstName.text,
+            lastName: _contactLastName.text,
+            phoneNumber: _contactPhoneNumber.unmasked,
+            avatar: _contactPhoto);
+        Navigator.pop(context);
+      } else {
+        _handleInvalidPermissions(permissionStatus);
+      }
+    }
+  }
+
+  /// ask for permission and the navigates to another screen as specified by [routeName]
+  Future<void> _askPermissions(String? routeName) async {
+    PermissionStatus permissionStatus = await _getContactPermission();
+    if (permissionStatus == PermissionStatus.granted) {
+      if (routeName != null) {
+        Navigator.of(context).pushNamed(routeName);
+      }
```

Comment:
What is this for ?
Ask permission should perform only one function, single responsibility. 
Don't add routing in the same